beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -5878,6 +5876,29 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
             .flatten()
     }
 
+    /// The epoch since which we cater blob data upon a request 'ByRoot'.
+    /// `None` if the `Eip4844` fork is disabled.
+    pub fn data_availability_boundary_by_root_rpc_request(&self) -> Option<Epoch> {
```

Comment:
Can we change this to call `data_availability_boundary` so we have a single method with the boundary logic

`self.data_availability_boundary().map(|boundary|std::cmp::max(boundary, self.canonical_head.cached_head().finalized_checkpoint().epoch))`